word_class.rb

```diff
@@ -0,0 +1,215 @@
+        puts "YOU WON!".colorize(:white)
+        exit
+      else
+        new_image = Cactus.new
+        puts new_image.show_ascii(6)
+        puts "You lost... the correct answer was\n".colorize(:white)
+        puts " " * 16 + "#{@word.upcase.colorize(:white)} \n\n"
+        exit
+      end
+    end
+  end
+
+#makes sure the whole word guessed does not have numbers in it
+  def acceptable_whole_word(whole_word_guess)
+    while is_a_letter?(whole_word_guess) == nil
+      puts "Say what? Please enter a word.".colorize(:red)
+      whole_word_guess = gets.chomp.downcase
+    end
+  end
+
+#makes sure the user input is a y or n
+  def y_or_n(input)
+    acceptable_answers = ["Y","N","y","n"]
+    while !acceptable_answers.include?(input)
+      puts "Woah woah woah. I only accept Y or N. Try again.".colorize(:red)
+      input = gets.chomp
+    end
+  end
+
+  #add_guesses takes the user's guess as input, then adds it to an array
+  #containing all the letters that have been guessed.  It displays that array.
+  def add_guess
+    @used_guesses << @current_guess
+    return @used_guesses
+  end
+
+  #update_gameboard: In the case of a correct guess, it add the guessed letter
+  #to the gameboard.  In the case of an incorrect guess, it adds 1 to the
+  #guess_count.  It then outputs the gameboard.
+  def update_gameboard
+    include_letter?
+    if @guess_ind.length != 0
+      puts "\n" + " " * 16 + "CORRECT!".colorize(:green)
+      @guess_ind.each do |i|
+        @gameboard[i] = @used_guesses[-1]
+      end
+    else
+      puts "\n" + " " * 16 + "INCORRECT".colorize(:red)
+      update_guess_count
+    end
+    output_gameboard
+  end
+
+  #output_gameboard flattens the gameboard array for better viewing
+  def output_gameboard
+    puts " " * 16 + @gameboard.join(" ").colorize(:white)
+  end
+
+  #include_letter? returns an array with the index of all instances of the
+  #guessed letter.
+  def include_letter?
+    @guess_ind = (0 ... @word.length).find_all { |i| @word[i,1] == @used_guesses[-1]}
+    return @guess_ind
+  end
+
+  #gameboard_complete checks if the game has been won (there are no blanks left)
+  def gameboard_complete
+    !@gameboard.include?("_")
+  end
+
+  #update_guess_count adds 1 to the guess_count
+  def update_guess_count
+    @guess_count += 1
+  end
+
+  #game_status checks if the game has been won, lost, or is still ongoing.
+  #if the game has been won, or lost, the game exits, otherwise it keeps going.
+  def game_status
+    if  gameboard_complete
+      puts "YOU WON!".colorize(:white)
+      exit
+    elsif @guess_count >= 6
+      new_image = Cactus.new
+      puts new_image.show_ascii(6)
+      puts "You lost... the correct answer was\n".colorize(:white)
+      puts " " * 16 + "#{@word.upcase.colorize(:white)} \n\n"
+      exit
+    end
+  end
+
+  #turn combines many of the methods in this class to encaspulate a single round
+  #of the game
+  def turn
+    know_the_word
+    obtain_guess
+    add_guess
```

Comment:
These method names are excellent, they let the reader know exactly what happens over the course of a turn without having to dig through a bunch of code.